staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -237,5 +237,40 @@ func (c *RESTClient) Delete() *Request {
 
 // APIVersion returns the APIVersion this RESTClient is expected to use.
 func (c *RESTClient) APIVersion() schema.GroupVersion {
-	return c.content.GroupVersion
+	return c.content.GetClientContentConfig().GroupVersion
+}
+
+type requestClientContentConfigProvider struct {
+	base    ClientContentConfig
+	tripped atomic.Bool
```

Comment:
give me a better name. If true, what is supposed to happen?  Is this `forceContentTypeToJSON`?

Reply:
Commented and renamed to `sawUnsupportedMediaTypeForCBOR` to describe how its value should be interpreted.